File: bridge/bridge-zero-android/app/src/main/java/arc/omega/bridgezero/RuntimeEventBus.kt
package arc.omega.bridgezero

import kotlinx.coroutines.flow.MutableSharedFlow
import kotlinx.coroutines.flow.SharedFlow
import kotlinx.coroutines.flow.asSharedFlow

sealed interface RuntimeStoreEvent {
    data class Status(val value: RuntimeStatus) : RuntimeStoreEvent
    data class DiagnosticsChanged(val value: Diagnostics) : RuntimeStoreEvent
    data class Telemetry(val value: RuntimeTelemetry) : RuntimeStoreEvent
    data class Session(val id: String?) : RuntimeStoreEvent
    data class Chronicle(val entries: List<ChronicleEntry>) : RuntimeStoreEvent
    data class RuntimeStreaming(val value: RuntimeStreamingSignal) : RuntimeStoreEvent
    data class Stream(val value: StreamEvent) : RuntimeStoreEvent
}

class RuntimeEventBus {
    private val mutableEvents = MutableSharedFlow<RuntimeStoreEvent>(extraBufferCapacity = 64)
    val events: SharedFlow<RuntimeStoreEvent> = mutableEvents.asSharedFlow()

    fun publish(event: RuntimeStoreEvent) {
        mutableEvents.tryEmit(event)
    }
}

File: bridge/bridge-zero-ios/Sources/RuntimeEventBus.swift
import Combine

enum RuntimeStoreEvent {
    case status(RuntimeStatus)
    case diagnostics(Diagnostics)
    case telemetry(RuntimeTelemetry)
    case session(String?)
    case chronicle([ChronicleEntry])
    case runtimeStreaming(RuntimeStreamingSignal)
    case stream(StreamEvent)
}

final class RuntimeEventBus {
    let events = PassthroughSubject<RuntimeStoreEvent, Never>()

    func publish(_ event: RuntimeStoreEvent) {
        events.send(event)
    }
}
